lib/initialize.ts

```diff
@@ -22,8 +22,23 @@ export default function createInitializer(
 
   return function init(
     initCb: (sdk: KnownSDK, customSdk: any) => any,
-    { makeCustomApi }: { makeCustomApi?: Function } = {}
+    {
+      makeCustomApi,
+      supressIframeWarning
+    }: { makeCustomApi?: Function; supressIframeWarning?: boolean } = {
+      supressIframeWarning: false
+    }
   ) {
+    if (!supressIframeWarning && currentWindow.self === currentWindow.top) {
+      console.error(`Cannot use ui-extension-sdk outside of Contenful:
+
+It appears that you are trying to use the ui-extension-sdk outside of Contentful
+
+In order for the ui-extension-sdk to function correctly, your app needs to be run in an iframe in the Contentful webapp
+
+Learn more about local development with the ui-extension-sdk here:
+  https://www.contentful.com/developers/docs/extensibility/ui-extensions/faq/#local-development-using-the-sdk`)
+    }
```

Comment:
I am not 100% sure about the wording. Content wise it's fine. Maybe add some periods after the sentences.